web/app/components/dropdown/dropdown.jsx

```diff
@@ -0,0 +1,81 @@
+import { h, Component } from 'preact';
+
+import Button from '../button';
+
+export default class Dropdown extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isActive: props.isActive || false,
+    };
+
+    this.onClickTitle = this.onClickTitle.bind(this);
+    this.onOutsideClick = this.onOutsideClick.bind(this);
+  }
+
+  bindRef(field) {
+    return (ref) => this[field] = ref;
+  }
+
+  onClickTitle() {
+    this.setState({
+      isActive: !this.state.isActive
+    });
+
+    if (this.props.onClickTitle) {
+      this.props.onClickTitle();
+    }
+  }
+
+  onOutsideClick(e) {
+    if (!this._dropdown.contains(e.target) && e.target !== this._dropdown) {
```

Comment:
Are you sure that you need the second condition here? `contains` checks the node itself, doesn't it?

https://developer.mozilla.org/en-US/docs/Web/API/Node/contains

Reply:
Nope, `e.target` could be any children html element too